### DISCUSSION TITLE
How best to use within adventure game with multiple location, character, objects, etc.?

### DISCUSSION THREAD

**Discussion by Liquidream** [inkle/ink#847] [Game integration help]
Hi all,

I'm new to Ink, so far it seems like Ink makes the most sense to use (rather than re-inventing the "narrative logic" wheel, so to speak). 

The game/engine I'm making is in Javascript/Typescript (so I'll be using the inkjs fork), but I believe my questions are more general to the greater world of Ink.

So far, the only real examples I've seen online for Ink usage has all been with one ink file driving the story.  But I'm curious what recommendations there might be on how best to use Ink within a large Point+Click adventure game (consisting of multiple locations, characters, objects, etc.).  Where you could interact with one character, finish talking, pickup a few objects, then move to another scene and interact with a different character (or perhaps the same character that followed you), etc.. 
As I can't really imagine building everything inside a massive .ink file.

For example:
- **Is it typical to have multiple .ink files - potentially split by scene/character/object?** 
- **If so, what's the recommended way to make them work with each other?** 
- **Can I still maintain a global state across the files within a single story?** (or would it be multiple?)
   - (e.g. if I change the variable/state of given character/object, I'd probably need to query it from another char/objs's ink file)
- Can I also use ink in this way to handle the kind of events I'm used to, such as `location.on_enter`/`on_exit` ?
(or is there another way I should be thinking about such event logic?)

Apologies if these are rather basic questions (+I'm probably trying to run before I can walk). 
Either way, I'd very much appreciate any advice you can give.

Thanks in advance! 🤓

**Comment by PeterJJonas** [comment]
Hi Liquidream,

I am just setting up my environment (Inky + Unity), so I am not an expert, but I can change variables, call Unity functions, and even outsource the logic to Unity and address entry points of the story. With that, I can use more Ink files, combine them, and pass variables between them.
This is Unity and Inky, but I think you can do the same with JavaScript/TypeScript.
I hope this helps.

Have a great one,
PeterJJonas

**Comment by Liquidream** [comment]
Hi @peterjjonas,

Thanks for the suggestions, much appreciated.

I recently stumbled across a great playlist of someone doing similar integration with Unity, and it's given me some ideas of how this could hopefully translate to JS/TS.

(In case you've not seen them before, [here's the playlist](https://www.youtube.com/playlist?list=PL3viUl9h9k78KsDxXoAzgQ1yRjhm7p8kl))

Thx again and all the best with ur project! 🤓

**Reply by PeterJJonas** [reply]
Hi @Liquidream,

Thank you for the playlist; I haven't seen it before.

I wish you the best with your project, too!
